packages/stripe-adapter/test/adapter.test.ts

```diff
@@ -16,209 +16,146 @@
+      fetchedInstancesTypes = (await fetchInstances()).map(i => i.elemID.typeName)
+    })
 
-      const taxRate = elements.filter(isInstanceElement).find(e => e.elemID.getFullName() === 'stripe.tax_rate.instance.txr_1IcrqNHipyrr1EYiTpN0iuVj')
-      expect(taxRate).toBeDefined()
-      expect(taxRate?.value).toEqual(expect.objectContaining({ id: 'txr_1IcrqNHipyrr1EYiTpN0iuVj' }))
+    // TODO - Update the mock replies file using Netta's script and remove the filter
+    test.each(SINGULAR_TYPES.filter(typeName => typeName !== 'product'))(
+      '%s',
+      typeName => {
+        expect(fetchedInstancesTypes).toContain(typeName)
+      },
+    )
+  })
 
+  describe('custom config', () => {
+    describe('include types', () => {
+      const INCLUDE_TYPES = ['coupons', 'tax_rates']
+      const SINGULAR_INCLUDE_TYPES = INCLUDE_TYPES.map(t => pluralToSingularTypes[t])
 
-      const webhookEndpoint = elements.filter(isInstanceElement).find(e => e.elemID.getFullName() === 'stripe.webhook_endpoint.instance.we_1IcrkHHipyrr1EYiSGhlscV7')
-      expect(webhookEndpoint).toBeDefined()
-      expect(webhookEndpoint?.value).toEqual(expect.objectContaining({ id: 'we_1IcrkHHipyrr1EYiSGhlscV7' }))
-    })
-  })
+      let fetchedInstancesTypes: Set<string>
 
-  describe('type overrides', () => {
-    it('should fetch only the relevant types', async () => {
-      const { elements } = await adapter.operations({
-        credentials: new InstanceElement(
-          'config',
-          accessTokenCredentialsType,
-          { token: 'aaa' },
-        ),
-        config: new InstanceElement(
+      beforeAll(async () => {
+        const config = new InstanceElement(
           'config',
           configType,
           {
             [FETCH_CONFIG]: {
-              includeTypes: ['prices'],
+              includeTypes: INCLUDE_TYPES,
             },
             [API_DEFINITIONS_CONFIG]: DEFAULT_API_DEFINITIONS,
           }
-        ),
-        elementsSource: buildElementsSourceFromElements([]),
-      }).fetch({ progressReporter: { reportProgress: () => null } })
-      expect(elements.filter(isInstanceElement)).toHaveLength(1)
-      expect(elements.filter(isInstanceElement).map(e => e.elemID.getFullName())).toEqual(['stripe.price.instance.price_1Ik5jyHipyrr1EYiTX75FImK'])
+        )
+        const instances = await fetchInstances(config)
+        fetchedInstancesTypes = new Set(instances.map(e => e.elemID.typeName))
+      })
+
+      describe('fetches included types', () => {
+        test.each(SINGULAR_INCLUDE_TYPES)(
+          '%s',
+          includedType => {
+            expect(fetchedInstancesTypes).toContain(includedType)
+          }
+        )
+      })
+
+      test('doesn\'t fetch additional types', async () => {
+        const additionalTypes: string[] = Array.from(fetchedInstancesTypes)
+          .filter(fetchedType => !SINGULAR_INCLUDE_TYPES.includes(fetchedType))
+        expect(additionalTypes).toBeEmpty()
+      })
```

Comment:
1. IMO it's clearer to combine those tests into one describe (but it's just my opinion):
```suggestion
      describe('fetches only included types', () => {
        test.each(SINGULAR_INCLUDE_TYPES)(
          '%s',
          includedType => {
            expect(fetchedInstancesTypes).toContain(includedType)
          }
        )
        const additionalTypes: string[] = Array.from(fetchedInstancesTypes)
          .filter(fetchedType => !SINGULAR_INCLUDE_TYPES.includes(fetchedType))
        expect(additionalTypes).toBeEmpty()
      })
```

2. also, we usually use `it` instead of `test`, just for the sake of consistency :)

Reply:
v

Reply:
1. This is not possible, Everything under describe should sit under a test :(

Reply:
1. I mean putting it all under an `it` which is under the `describe` (maybe wasn't clear enough in the suggestion)

Reply:
You're sharp! Had unnecessary extra description. Changed the outer scope description